src/Rules/Properties/PropertyInClassRule.php

```diff
@@ -76,6 +76,17 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
+		if ($node->isReadOnly()) {
+			if ($node->hasHooks()) {
```

Comment:
You could move both conditions into a single if and avoid a nesting level?

Reply:
In theory I can, but I wanted to be consistent with how we handled the abstract ones.

Reply:
Ah OK, was just curious about the reason and didn't see that.

Reply:
Of course please check first whether PHP allows it or not, I might be wrong.